packages/apollo-codegen-typescript/src/codeGeneration.ts

```diff
@@ -223,13 +231,24 @@ export class TypescriptAPIGenerator extends TypescriptGenerator {
     this.scopeStack = [];
   }
 
-  fileHeader() {
+  fileHeader(tagName?: string) {
```

Comment:
This is a bit confusing for me, since specifying `tagName` by itself does not actually guarantee that the `import` will be generated. Could this function be adjusted to always generate the `import ${tagName}` if `tagName` is defined and then also generate the `import codeGenerationModule` if that config value is specified? This would have to come with a change above to call `generator.fileHeader(codeGenerationModule && "gql")`.

Also, I think the logic to generate imports merits a function of its own, since it's less of a header and fits in more with the rest of the types being generated.

Reply:
Ah, I can see how that'd be confusing. I've split the import generation logic off into `generator.fileImports(tagName: string, codeGenerationModule: string)` and changed it so it'd always generate the `import ${tagName}` if `tagName` is defined.